lib/pg_saurus/connection_adapters/postgresql_adapter/function_methods.rb

```diff
@@ -97,6 +105,21 @@ def parse_function_language(function_str)
   end
   private :parse_function_language
 
+  # Retrieve the volatility of the function: volatile, stable, or immutable
+  # @return [Symbol]
+  def parse_function_volatility(function_str)
+    rows = function_str.split("\n")
+    lang_index = rows.index { |line| line =~ /LANGUAGE/ }
+    def_index = rows.index { |line| line =~ /AS \$function\$/ }
+
+    if lang_index && def_index && def_index - lang_index == 2
+      :"#{rows[def_index - 1].strip.split(" ").first.downcase}"
```

Comment:
This is a brittle means of locating the volatility token. It will likely break when support for other features of `CREATE FUNCTION` is added.